### ISSUE TITLE
Adding component to bad pool - Should throw exception

### ISSUE THREAD

**Issue by LeRondPoint** [sschmid/Entitas-CSharp#215]
Given the following components :

```
[PoolA]
[SingleEntity]
class AComponent : IComponent
{ whatever }

[PoolB]
class BComponent : IComponent
{ whatever }
```

Both components end up having the save Id (let's say 1).  If you add BComponent to an entity in PoolA, you will will get PoolA.hasA to true but your PoolA.aEntity doesn't hold the component (is had a component of type BComponent) and PoolA.a throws an exception.  The best would be to throw an exception if a component is added to an entity in a pool where it doesn't belong...

**Comment by T2RKUS** [comment]
Is't that what the new type safety feature branch would address. 
PoolAEntity class inherits from Entity which acts like a gate for managing what components a pool would have. so you'd do 
``
var poolA = new Pool<PoolAEntity>();
PoolA.CreateEntity().AddPosition();
``
so the generated extension code is moved there.

**Comment by LeRondPoint** [comment]
Possibly, I wasn't aware of that branch!

**Comment by sschmid** [comment]
Closed by #266
